.github/workflows/ci.yml

```diff
@@ -66,8 +65,35 @@ jobs:
       with:
         action: ${{ matrix.actions.action }}
 
-  test:
-    name: Test
+  test-codegen:
```

Comment:
Can we add a longer form description here to explain the grouping? (Test of white label smithy codegen behavior, no aws sdk dependency, etc.)

We may be able to eventually go one step further and not run these at all for changes in the aws directory